File: javascript/2553.js
/**
 * @param {number[]} n
 * @return {number[]}
 */
var separateDigits = function (n) {
    let r = [];
    for (let i of n) {
        let s = 1, t = parseInt(i / 10);
        while (t > 0) {
            s *= 10;
            t = parseInt(t / 10);
        }
        while (s > 0) {
            r.push(parseInt(i / s));
            i %= s;
            s = parseInt(s / 10);
        }
    }
    return r;
};

File: java/2553.java
class Solution {
    public int[] separateDigits(int[] n) {
        int p = 0;
        for (int i : n)
            while (i > 0) {
                p++;
                i /= 10;
            }
        int r[] = new int[p];
        p = 0;
        for (int i : n) {
            int s = 1, t = i / 10;
            while (t > 0) {
                s *= 10;
                t /= 10;
            }
            while (s > 0) {
                r[p++] = i / s;
                i %= s;
                s /= 10;
            }
        }
        return r;
    }
}
